fmn/consumer/backend.py

```diff
@@ -152,22 +152,33 @@ def read(queue_object):
 
     if function:
         print("Got a function call to %s" % function)
-        if function == 'handle':
-            backend = backends[data['backend']]
-            backend.handle(session,
-                           data['recipient'],
-                           data['msg'],
-                           data['streamline'])
-            ch.basic_ack(delivery_tag=method.delivery_tag)
-        elif function == 'handle_batch':
-            backend = backends[data['backend']]
-            backend.handle_batch(session,
-                                 data['recipient'],
-                                 data['queued_messages'])
+        try:
+            if function == 'handle':
+                backend = backends[data['backend']]
+                backend.handle(session,
+                               data['recipient'],
+                               data['msg'],
+                               data['streamline'])
+                ch.basic_ack(delivery_tag=method.delivery_tag)
+            elif function == 'handle_batch':
+                backend = backends[data['backend']]
+                backend.handle_batch(session,
+                                     data['recipient'],
+                                     data['queued_messages'])
+                ch.basic_ack(delivery_tag=method.delivery_tag)
+            else:
+                print("Unknown function")
+            return
+        except Exception:
+            logging.error('Message failed, requeueing', exc_info=True)
+            ch.basic_publish(exchange='',
```

Comment:
I think this will work fine for now, but in the future I'd like to have a better method of dealing with undelivered messages. This will result in a message that gets re-queued for eternity/when a bugfix is applied.

Better retry logic and graceful handling of missing external services in combination with, say, a [dead letter exchange](https://www.rabbitmq.com/dlx.html) to record the offending messages might be good. Then maybe a service that takes dead letters and sends a generic alert to users with a datanommer link? Not sure if that's feasible at all, but it's food for thought.